gptme/tools/mcp_adapter.py

```diff
@@ -0,0 +1,159 @@
+from logging import getLogger
+from collections.abc import Callable
+import json
+
+from ..message import Message
+from ..mcp.client import MCPClient
+from .base import Parameter, ToolSpec
+
+# Define ConfirmFunc type directly to avoid circular imports
+ConfirmFunc = Callable[[str], bool]
+
+logger = getLogger(__name__)
+
+# Add type annotation for tool_specs
+tool_specs: list[ToolSpec] = []
+
+
+# Function to create MCP tools
+def create_mcp_tools(config) -> list[ToolSpec]:
+    """Create tool specs for all MCP tools from the config"""
+
+    tool_specs: list[ToolSpec] = []
+    servers = {}
+
+    # Skip if MCP is not enabled
+    if not config.mcp.enabled:
+        return tool_specs
+
+    # Initialize connections to all servers
+    for server_config in config.mcp.servers:
+        try:
+            client = MCPClient(config=config)
+
+            # Connect to server
+            tools, session = client.connect(server_config.name)
+
+            # Store the connection
+            servers[server_config.name] = {
+                "client": client,
+                "tools": tools,
+                "session": session,
+            }
+
+            # Create tool specs for each tool
+            for mcp_tool in tools.tools:
+                # Extract parameters
+                parameters = []
+                # Check if the tool has inputSchema with properties
+                if (
+                    hasattr(mcp_tool, "inputSchema")
+                    and isinstance(mcp_tool.inputSchema, dict)
+                    and "properties" in mcp_tool.inputSchema
+                ):
+                    required_params = mcp_tool.inputSchema.get("required", [])
+                    for param_name, param_schema in mcp_tool.inputSchema[
+                        "properties"
+                    ].items():
+                        parameters.append(
+                            Parameter(
+                                name=param_name,
+                                description=param_schema.get("description", ""),
+                                type=param_schema.get("type", "string"),
+                                required=param_name in required_params,
+                            )
+                        )
+
+                # Add example usage in the correct format
+                example = {
+                    param.name: f"<{param.type}>"
+                    for param in parameters
+                    if param.required
+                }
+                example_str = json.dumps(example, indent=2)
+
+                tool_spec = ToolSpec(
+                    name=f"{server_config.name}.{mcp_tool.name}",
```

Comment:
I think this may fix the issue you had with tools not being detected/used correctly by the existing tooluse detection logic (it splits on spaces).

```suggestion
                    name=f"{server_config.name.replace(" ", "_")}.{mcp_tool.name}",
```